app/controllers/user_badges_controller.rb

```diff
@@ -37,15 +39,16 @@ def username
     user_badges = user.user_badges
 
     if params[:grouped]
-      user_badges = user_badges.group(:badge_id)
-        .select(UserBadge.attribute_names.map { |x| "MAX(#{x}) AS #{x}" }, 'COUNT(*) AS "count"')
+      user_badges = user_badges.group(:badge_id).select_for_grouping
     end
 
     user_badges = user_badges.includes(badge: [:badge_grouping, :badge_type])
       .includes(post: :topic)
       .includes(:granted_by)
 
-    render_serialized(user_badges, DetailedUserBadgeSerializer, root: :user_badges)
+    render_serialized(user_badges, DetailedUserBadgeSerializer, root: :user_badges, meta: {
+                                                                  max_favorites: MAX_FAVORITES,
+                                                                })
```

Comment:
These lines look a bit weird. Maybe something like this

```suggestion
    render_serialized(user_badges, DetailedUserBadgeSerializer, 
      root: :user_badges, 
      meta: { max_favorites: MAX_FAVORITES }
    )
```

Or even like this?

```suggestion
    render_serialized(
      user_badges, 
      DetailedUserBadgeSerializer, 
      root: :user_badges, 
      meta: { max_favorites: MAX_FAVORITES }
    )
```

Reply:
Yeah you're right it does like a little weird. I went with the second option because I've seen it elsewhere in the codebase. Thanks for pointing that out!